### ISSUE TITLE
Fan speed not increasing with high temp.

### ISSUE THREAD

**Issue by notasheepl** [linux-surface/linux-surface#1343]
I have an issue running Linux on the surface laptop 4 AMD. No matter what scaling driver I use, it always boosts to 4.0 GHz and instantly reaches 90°C but the fan won't turn up. This also happens in ubuntu with a TLP.

The following is happening in ubuntu with pstate-epp driver and TLP with the power profiles stopped. Sometimes things will come to a crawl, the CPU frequency won't go over 400mhz and the fan will be stuck at a very high rpm. The temp drops instantly but it gets stuck in that state for a while even though temperatures are no longer a problem.

In windows temps also get high but I can feel and hear the fan working. If I set CPU limit to 99% and set power profile to recommended, turbo boost stops working and frequency stays around 2.2ghz which is fine for most tasks and temperatures are around 60-70°C.

**Comment by varcharfoobar** [comment]
A quick little update, I played around with TLP (no luck with this so removed it from my system) but I noticed there is a slight improvement with "surface-control" being installed. 

- surface-control
- power-profiles-daemon (usually installed by default)

I need more time to test this out but I found the balanced profile to be pretty good. If you run GNOME or KDE, you can essentially set the profile to "balanced" and it should change the "surface status" to "balanced" as well.

Here is an output:

```
berk@surface:~$ sudo surface status
Platform Profile: low-power [balanced] balanced-performance performance
```

The frequencies are very reasonable now running various browsers, Android Studio and utility applications in the background. I also ran a stress test with "stress-ng" for about two mins and it was fine. But I'll have to do more extensive testing.

**Comment by notasheepl** [comment]
TLP is not recommended according to readme.
https://github.com/linux-surface/linux-surface?tab=readme-ov-file#additional-information

This doesn't just happen under heavy load for me.
This behaviour also happens when under a light load like watching a 480p video in a web browser or even in the GRUB menu.
I feel uncomfortable having such high temps on a small device, especially with the lid closed so mainly use windows on it.
Last I tried Fedora 39 KDE with only the packages listed in the wiki (no secure boot), and it was still exhibiting this issue. No fan, high temp, max boost.

**Comment by notasheepl** [comment]
![Screenshot_20240312_214801.png](https://github.com/linux-surface/linux-surface/assets/90622241/c4186969-ae94-49a3-86bd-0c0753275dc6)

![Screenshot_20240312_212116.png](https://github.com/linux-surface/linux-surface/assets/90622241/01375aba-53e2-48fa-a00a-86ba5220b053)

This is on battery. Same behaviour on AC. Stays pegged at 90°C.
The fan is still not working.

**Comment by varcharfoobar** [comment]
Yeah, I can confirm too now. Same issue :( Very sad, I've had to put Windows 11 on the Surface which does work well but Windows as a desktop just feels a slow compared to GNOME/KDE. Maybe one day the frequency/throttling/cooling will be all resolved. 

I've also had to disable "iommu" and "amd_iommu" in order to boot into Fedora (edit Grub config before boot up). There was no touch which I was cool with but seeing the computer throttle and simply stuck at 4 GHz + fan doesn't increase as you said, it was a bit unusable for me.

**Comment by notasheepl** [comment]
Have you tried auto-cpufreq? Last time I tried it didn't help but you might have better luck.
https://github.com/AdnanHodzic/auto-cpufreq

**Comment by varcharfoobar** [comment]
No, haven't tried that yet -- but I also can't right now since I've set up everything back onto Windows 11 unfortunately (got a ThinkPad for my Linux needs).

But if you have tried it, I'd imagine it's much of the same experience since we both have the same hardware. If you are still running Fedora Plasma, maybe you can try and uninstall the "power-profiles-daemon" package and then just have "auto-cpufreq" active (maybe the two were clashing?)

Up to you of course if you want to try that (it may also require a logout/restart) just to be sure.

**Comment by berkiyo** [comment]
Just a quick update.

I'm back on the SL4A now running Fedora 40 and installed just the following packages:

- surface-control (left it on defaults, so "balanced")
- auto-cpufreq

All I ran was `sudo auto-cpufreq --install` and then simply monitor the temperatures.

When running `sudo auto-cpufreq --stats` and pushing the GPU with Minecraft + Shaders (just for testing), the temps sit around 75-90 C.

So it's quite high unfortunately. When I put my ear next to the fans, there is no spin up so the computer is either running the fans at a low RPM or, they are completely off. So essentially running fanless like a MacBook Air.

When typing "sensors" or using the "Vitals" extension in GNOME, there is also no fan data which could suggest the fan is not detected at all. :( 

---

I just keep the workload light on this laptop due to this thermal issue but hopefully one day we can have this issue resolved. 

PS: I also noticed the CPU freq is constantly quite high (alternating between boost and base block, but never going below 2.0 GHz).

**Comment by berkiyo** [comment]
Another update folks! With some good news.

So the fan does work however, it takes a while for it to turn on.

The fan itself however, runs at a very low RPM but enough to keep the CPU below 90C when under load. 

When cool, the CPU will turbo to 4 GHz and then clock down after 20-30 seconds to about 3.4-3.6 GHz.

Running `stress` for 2 minutes across 12 threads, temps didn't exceed 90 C and fan turned on. Pretty stable but not perfect of course.

That being said, I am happy with how it is set up now. Here is my final set up:

- surface-kernel (latest)
- surface-control (running "balanced-performance")
- auto-cpufreq
- Fedora 40 + GNOME @ 100% scaling

So overall, working well enough for my needs as a portable Linux machine. 😉

**Comment by Eilermoon** [comment]
Hi! This was super helpful to me. I'm running an intel Surface Laptop 4 that has had constant thermal issues and hard throttling when running any super light games or video calls/streams. I honestly wasn't even sure this had a fan in it until swapping the profile over to performance like you mentioned above.

Is there a built-in way to automatically change over to these profiles based on performance demand or is the only solution to command this manually or write my own script to do so? I'd rather not stay in performance mode constantly. Thanks!
